src/coreclr/vm/eetwain.cpp

```diff
@@ -4712,16 +4714,41 @@ bool EECodeManager::EnumGcRefs( PREGDISPLAY     pContext,
             {
                 _ASSERTE(willContinueExecution);
 
+#ifdef FEATURE_EH_FUNCLETS
+                // Funclets' frame pointers(EBP) are always restored so they can access to main function's local variables.
+                // Therefore the value of EBP is invalid for unwinder so we should use ESP instead.
+                // See UnwindStackFrame for details.
+                if (pCodeInfo->IsFunclet())
+                {
+                    PTR_CBYTE methodStart = PTR_CBYTE(pCodeInfo->GetSavedMethodCode());
+                    TADDR baseSP = ESP;
+                    // Set baseSP as initial SP
+                    baseSP += GetPushedArgSize(&info, table, curOffs);
+                    // 16-byte stack alignment padding (allocated in genFuncletProlog)
+                    // Current funclet frame layout (see CodeGen::genFuncletProlog() and genFuncletEpilog()):
+                    //   prolog: sub esp, 12
```

Comment:
This padding is Linux x86 specific. This should be under `STACK_ALIGN` or `UNIX_X86_ABI` condition, both here and in the JIT.

Reply:
Would it make sense to handle that change separately? I would feel more confident doing it once all the bits are in place to run the smoke tests.